internal/command/views/cloud.go

```diff
@@ -4,53 +4,192 @@
 package views
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"strings"
 	"time"
+
+	"github.com/hashicorp/terraform/internal/command/arguments"
+	"github.com/hashicorp/terraform/internal/tfdiags"
 )
 
-// CloudHooks provides functions that help with integrating directly into
-// the go-tfe tfe.Client struct.
-type CloudHooks struct {
-	// lastRetry is set to the last time a request was retried.
+// The Cloud view is used for operations that are specific to cloud operations.
+type Cloud interface {
+	RetryLog(attemptNum int, resp *http.Response)
+	Diagnostics(diags tfdiags.Diagnostics)
+	Output(messageCode CloudMessageCode, params ...any)
+	PrepareMessage(messageCode CloudMessageCode, params ...any) string
+}
```

Comment:
I think `RetryLog` is the only function here that is called externally to this package right? We don't need to expose the other functions here as nothing is calling them. I'd make `Diagnostics`, `Output` and `PrepareMessage` private functions available only the human and json implementations instead of exposing them via the public interface.

Reply:
I would definitely like the cloud backend to begin leveraging this interface as opposed to `c.UI`. `Diagnostics` is fine to expose publicly. `Output()` and `PrepareMessage()`, the way they're currently implemented seem tightly coupled with `RetryLog`, so it might be necessary to decouple them or make them more specific.

Reply:
I've made `PrepareMessage` and `Output` private functions, available to the Human and JSON Cloud implementations.